### ISSUE TITLE
[BUG] Capitalization has changed when BlobTraits are serialized into request URI parameters for List Blob requests

### ISSUE THREAD

**Issue by egraff** [Azure/azure-sdk-for-net#22206]
**Describe the bug**
I'm a bit unsure whether this should actually be classified as a bug, or whether the behavior is acceptable.

Basically, I've noticed that after upgrading from Azure.Storage.Blobs 12.8.4 -> 12.9.1, the URI parameters that are generated when calling `BlobContainerClient.GetBlobsAsync()` and specifying some traits (such as  `BlobTraits.Metadata`) are capitalized differently than before the upgrade.

Specifically, doing something like:
```csharp
AsyncPageable<BlobItem> asyncPageable = containerClient.GetBlobsAsync(BlobTraits.Metadata);
IAsyncEnumerator<BlobItem> enumerator = asyncPageable.GetAsyncEnumerator();
```
and then iterating the enumerator will generate List Blob requests. In version 12.8.4, the request URI would look something like `https://<server>/<account>/<container>?restype=container&comp=list&include=metadata`, whereas in version 12.9.1 the request URI looks like `https://<server>/<account>/<container>?restype=container&comp=list&include=Metadata` (notice the capital `M` in `Metadata`).

Again, I'm not sure if this is a big deal, but the difference made some of my integration tests fail, because I'm mocking the Azure Storage REST API and did a case-sensitive string comparision (the official [api docs](https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs) does not mention anything about how capitalization is handled, although I suspect that the implementation might be permissive enough to handle this difference?).

I thought I would report it nonetheless, in case this change was not intentional.

**Expected behavior**
`GetBlobsAsync(BlobTraits.Metadata)` should generate request URIs containing `include=metadata` as query parameter.

**Actual behavior (include Exception or Stack Trace)**
`GetBlobsAsync(BlobTraits.Metadata)` generates request URIs containing `include=Metadata` as query parameter.

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.9.1

**Comment by egraff** [comment]
The change seems to come from https://github.com/Azure/azure-sdk-for-net/commit/1814567d1f6cfb4905539e71d54377ab46ed6737.

In version 12.8.4, the file `Azure.Storage.Blobs/src/Generated/BlobRestClient.cs` contained a function `ListBlobsFlatSegmentAsync_CreateMessage` with code looking like

```csharp
if (include != null) { _request.Uri.AppendQuery("include", string.Join(",", System.Linq.Enumerable.Select(include, item => Azure.Storage.Blobs.BlobRestClient.Serialization.ToString(item)))); }
```

whereas in 12.9.1, the file `Azure.Storage.Blobs/src/Generated/ServiceRestClient.cs` contains a function `CreateListContainersSegmentRequest` with this code:
https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/storage/Azure.Storage.Blobs/src/Generated/ServiceRestClient.cs#L283-L286

and the `AppendQueryDelimited()` function simply calls `ToString()` on the elements of the collections, so the serialization extension method that was used previously is no longer used.

**Comment by amishra-dev** [comment]
URLs are supposed to be case-insensitive, can I safely assume that you are not hindered by it but just wanted to let us know?

**Comment by egraff** [comment]
> URLs are supposed to be case-insensitive

@amishra-dev excuse my straightforwardness, but I believe you are mistaken. This is not true in the general case, and I just confirmed that for Azure Storage, at least blob names are case _sensitive_. I cannot comment on whether the query parameter values are supposed to be case-sensitive or case-insensitive though, as I cannot find this documented anywhere for Azure Blob Storage.

> can I safely assume that you are not hindered by it but just wanted to let us know?

@amishra-dev to be honest, I haven't really tried version 12.9.1 against real Azure Blob Storage (I rolled back the NuGet upgrade after seeing some of my integration tests fail) 😛 I've just assumed that it might still work, despite the observed difference. 

However, I still think this _is_ a bug (albeit probably not a high-priority one _if_ the requests do still work as before), since relying on the string representation of enum members seems non-ideal, especially considering there is still code in the repo that handles the serialization properly (https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/storage/Azure.Storage.Blobs/src/Models/Internal/ListBlobsIncludeItemExtensions.cs), but which is now unused.
